### PR TITLE
Add watches on Image and ConfigMap

### LINKED ISSUE #17: Add Watches on secondary resources
Copying this [comment](https://github.com/openshift/cincinnati-operator/pull/11#pullrequestreview-395587302) over to an issue.

I think this needs two more Watches, both with a new handler.

1. Watch the Image resource. The [generation change predicate](https://github.com/kubernetes-sigs/controller-runtime/blob/1a11c4a/pkg/predicate/predicate.go#L134) would be a good idea to use here.
2. Watch ConfigMap resources in the openshift-config namespace. Might have to use a predicate to limit it to that namespace.

You'll need a new handler of the [EnqueueRequestsFromMapFunc](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/handler#EnqueueRequestsFromMapFunc) type. I think the logic will go like this:

```
If object is Image:
  return Requests for all Cincinnati instances
If object is ConfigMap:
  If ConfigMap is the one referenced from the Image:
    return Requests for all Cincinnati instances
```

That'll make sure that if either changes, it'll enqueue reconciliation for all Cincinnati resources, and then they can each react accordingly based on the code you've already written.

Here is [an example](https://github.com/openshift/cluster-api-provider-baremetal/blob/e8327d4/pkg/manager/wrapper/mapper.go) of a simple mapper and the [associated Watch](https://github.com/openshift/cluster-api-provider-baremetal/blob/6c1e3cf/pkg/manager/wrapper/wrapper.go#L53).

### PR DESCRIPTION
**Pull request by sacharya**
Fixes #17

### CODE REVIEW

**Comment by sacharya** [comment]
Not sure what is benefit of having the wrapper for the manager as opposed to adding the watches in the manager directly. I didn't remove it yet in case there are a good reason for it.

File: pkg/controller/cincinnati/mapper.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+func (m *mapper) Map(obj handler.MapObject) []reconcile.Request {
+	if cm, ok := obj.Object.(*corev1.ConfigMap); ok {
+		// There is already a watch on local configMap as a secondary resource
+		// This watch is for the source configMap in openshift-config namespace
+		if cm.Namespace != openshiftConfigNamespace {
+			return []reconcile.Request{}
+		}
+		image := &apicfgv1.Image{}
+		err := m.client.Get(context.TODO(), types.NamespacedName{Name: defaults.ImageConfigName, Namespace: ""}, image)
+		if err != nil {
```
[the remark this answers is not in the document]
Comment (sacharya) [reply pkg/controller/cincinnati/mapper.go:32]: Done.

**Comment by rthallisey** [comment]
nice job Sudarshan

**Pull request merged**